File: 2961_Double_Modular_Exponentiation/main.go
func getGoodIndices(variables [][]int, target int) []int {
	var ans []int

	for i, vars := range variables {
		a, b, c, m := int64(vars[0]), int64(vars[1]), int64(vars[2]), int64(vars[3])
		a = powAndMod(a, b, int64(10))
		a = powAndMod(a, c, m)
		if a == int64(target) {
			ans = append(ans, i)
		}
	}

	return ans
}

func powAndMod(base int64, exponent int64, mod int64) int64 {
	result := int64(1)
	for i := int64(0); i < exponent; i++ {
		result = result * base % mod
	}
	return result
}

File: 2961_Double_Modular_Exponentiation/main.py
class Solution:
    def getGoodIndices(self, variables: List[List[int]], target: int) -> List[int]:
        result = []
        for index in range(len(variables)):
            vars = variables[index]
            if (vars[0] ** vars[1] % 10) ** vars[2] % vars[3] == target:
                result.append(index)

        return result
